v2/tools/generator/internal/astmodel/allof_type.go

```diff
@@ -105,30 +105,33 @@ func (allOf *AllOfType) References() TypeNameSet {
 	return result
 }
 
-var allOfPanicMsg = "AllOfType should have been replaced by generation time by 'convertAllOfAndOneOf' phase"
+var allOfFailureMsg = "AllOfType should have been replaced by generation time by 'convertAllOfAndOneOf' phase"
 
 // AsType always panics; AllOf cannot be represented by the Go AST and must be
 // lowered to an object type
 func (allOf *AllOfType) AsType(_ *CodeGenerationContext) dst.Expr {
-	panic(errors.New(allOfPanicMsg))
+	panic(errors.New(allOfFailureMsg))
 }
 
 // AsDeclarations always panics; AllOf cannot be represented by the Go AST and must be
 // lowered to an object type
-func (allOf *AllOfType) AsDeclarations(_ *CodeGenerationContext, _ DeclarationContext) []dst.Decl {
-	panic(errors.New(allOfPanicMsg))
+func (allOf *AllOfType) AsDeclarations(
+	codeGenerationContext *CodeGenerationContext,
+	declContext DeclarationContext,
+) ([]dst.Decl, error) {
+	return nil, errors.New(allOfFailureMsg)
```

Comment:
Is this actually _expected_?

I don't think it is? Or maybe put another way, there's nothing in type configuration or otherwise that should cause this, right?

Should this stay a panic (even though it _could_ be an error now)?

Reply:
I've been applying the rubric to use errors for things that are most likely to be triggered by external factors, and panic for things most likely to be triggered by developer errors.

At this point, I'm fairly confident that our code is working properly. 

I figure the most likely case where things go wrong with an AllOf or OneOf type is if a Product Group does something new with their OpenAPI spec definitions that we haven't seen before.

Reply:
> At this point, I'm fairly confident that our code is working properly.

But, wouldn't it be a developer error if all `AllOf` types weren't removed by the time we get to the declarations stage of the pipeline?

I thought the contract we had was that `OneOf` and `AllOf` had to be turned into `ObjectType` before the end of the pipeline. I don't think it's possible for _any_ configuration to cause them to stick around? Wouldn't that be a developer error?

Reply:
I'm not thinking about ASO configuration - I'm thinking about what happens when someone uses an AllOf in a new way that we haven't seen before. That's an external factor, where I don't think we should panic - because external data shouldn't be trusted.

Reply:
That said, I'm not fervent on this, happy to change it back.

Reply:
> I'm not thinking about ASO configuration - I'm thinking about what happens when someone uses an AllOf in a new way that we haven't seen before.

You mean in our code or in the Swagger? If in the Swagger, I'm 90% sure that the problem won't come here, it will come from the `ConvertAllOfAndOneOfToObjects` stage (which does not call `AsDeclarations`). If you mean in our code, that's developer error right?

I've approved as I am also not fervent on this, but I do tend towards leaving this as a panic as I'm still not understanding what non-developer configuration could possibly cause this code to be hit. My assertion is that if this code is ever hit we have a code bug in the `ConvertAllOfAndOneOfToObjects` that somehow missed converting AllOf or OneOf types, which I don't think could happen unless somehow the visitor failed to visit them.

Also happy to chat and understand how I'm wrong about the above.

Reply:
Changed back to panic.